src/arizona_template_scanner.erl

```diff
@@ -0,0 +1,588 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright 2024 Arizona Framework <contact@arizonaframe.work>
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%%
+-module(arizona_template_scanner).
+-moduledoc false.
+
+%% API functions.
+-export([scan/2]). -ignore_xref([scan/2]).
+
+%% Types
+-export_type([token/0]).
+-export_type([anno/0]).
+-export_type([options/0]).
+
+-type state() :: #{
+    source => {function, {module(), atom()}} | {file, binary()} | none,
+    line => line(),
+    column => column(),
+    first_column => non_neg_integer(),
+    position => non_neg_integer()
+}.
+-opaque token() :: {open_tag, anno(), binary()}
+                 | {attr_key, anno(), binary()}
+                 | {attr_value, anno(), {text, binary()} | {expr, binary()}}
+                 | {bool_attr, anno(), binary()}
+                 | {close_tag, anno(), void | nonvoid}
+                 | {closing_tag, anno(), binary()}
+                 | {text, anno(), binary()}
+                 | {expr, anno(), binary()}.
+-opaque anno() :: location().
+-type location() :: {line(), column()}.
+-type line() :: non_neg_integer().
+-type column() :: non_neg_integer().
+-type options() :: #{
```

Comment:
Should this be `opaque`?

Reply:
`location`, `line`, and `column` are now opaque. `option` does not exist anymore.
https://github.com/arizona-framework/arizona/pull/143/commits/9c030c1b750e3eeb90fa8b4125a55d5d2fc8bc0a